modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java

```diff
@@ -1491,12 +1536,19 @@ private FileWriteHandle initNextWriteHandle(FileWriteHandle cur) throws IgniteCh
 
             boolean interrupted = false;
 
+            long nextSegmentId = cur.getSegmentId() + 1;
+
+            int segmentStatus;
+            if (useAsyncRollover && (segmentStatus = getSegmentStatus(nextSegmentId)) != SEGMENT_CLOSED)
```

Comment:
Are you sure that we shouldn't wait here, and fail immediately instead?
By the way, I think braces are missing for this `if`

Reply:
Unfortunately not. This method called during node start and node activation. In case of cluster deactivation segment status may be "SEGMENT_ACTIVE".